### PR TITLE
Add compilation instructions to the hacking howto

### LINKED ISSUE #2792: No compilation instructions
Version: i3 version 4.13-34-g7785e7b (2016-11-08, branch "master") © 2009 Michael Stapelberg and contributors

There is no instructions about compilation in i3's doc and/or readme file. I had to rely on the helpful guys on IRC (thanks again guys) which pointed me to https://github.com/Airblader/i3/wiki/Compiling-&-Installing

### PR DESCRIPTION
**Pull request by hwangcc23**
Fix #2792.

### CODE REVIEW

File: docs/hacking-howto [hunk 1]
```diff
@@ -990,6 +990,49 @@ New features are only found in the “next” branch. Therefore, if you are work
...
+
+(The autoreconf -fi step is unnecessary if you are building from a release tarball,
+ but shouldn’t hurt either.)
+
+==== Build system features
+
+* We use the AX_ENABLE_BUILDDIR macro to enforce builds happening in a separate
+  directory. This is a prerequisite for the AX_EXTEND_SRCDIR macro and building
+  in a separate directory is common practice anyway. In case this causes any
+  trouble when packaging i3 for your distribution, please let me know.
```
Comment (Airblader) [inline docs/hacking-howto]: s/let me know/open an issue/

File: docs/hacking-howto [hunk 2]
```diff
@@ -990,6 +990,49 @@ New features are only found in the “next” branch. Therefore, if you are work
...
+
+* “make check” runs the i3 testsuite.
+  You can still use ./testcases/complete-run.pl to get the interactive progress
```
Comment (Airblader) [inline docs/hacking-howto]: Can we make this refer to the page about the testsuite?

File: docs/hacking-howto [hunk 3]
```diff
@@ -990,6 +990,49 @@ New features are only found in the “next” branch. Therefore, if you are work
...
+* “make” will build manpages/docs by default if the tools are installed.
+  Conversely, manpages/docs are not tried to be built for users who don’t want
+  to install all these dependencies to get started hacking on i3.
+
+* non-release builds will enable address sanitizer by default. Use the
+  --disable-sanitizers configure option to turn off all sanitizers, and see
+  --help for available sanitizers.
+
+* Support for pre-compiled headers (PCH) has been dropped for now in the
+  interest of simplicitly. Maybe we can re-add it later.
```
Comment (Airblader) [inline docs/hacking-howto]: s/simplicitly/simplicity/

Comment (Airblader) [inline docs/hacking-howto]: Also let's rephrase the last sentence to »If you need support for PCH, please open an issue.«

**Comment by hwangcc23** [comment]
@Airblader Thanks for the review. It is revised.

**Pull request merged**